src/Export/ProductService.php

```diff
@@ -383,12 +395,14 @@ protected function getParentByCheapestVariant(ProductEntity $product): ProductEn
         // a cheaper product in its children.
         $cheapestPrice = $parent->getCurrencyPrice($currencyId);
         foreach ($children as $child) {
-            $price = $child->getCurrencyPrice($currencyId);
-            if (!$price) {
+            if (!$price = $child->getCurrencyPrice($currencyId)) {
                 continue;
             }
 
-            if ($price->getGross() < $cheapestPrice->getGross()) {
+            if (
+                $cheapestPrice->getGross() === 0.0 ||
```

Comment:
Shouldn't we `continue` when price=0?

Reply:
The cheapest price is defined before the `foreach`, to have some value to start with. The value is initially set to the parent price.
This check just makes sure, that the price is set with a value > 0, in case the main product has no price. At this place, all variants with a price of 0 are already filtered out, so no check needed.